### DISCUSSION TITLE
Two `@` when using mention

### DISCUSSION THREAD

**Discussion by Ni1kko** [DSharpPlus/DSharpPlus#1514] [General]
Two `@` when using mention from `voicestateupdateeventargs / userstate / channel / guild / everyonerole`
  
 ![alt text](https://i.imgur.com/ImG6252.png)
 
 ![2](https://i.imgur.com/iKlGHGg.png)
 
 ![3](https://i.imgur.com/lKVKIC0.png)

**Comment by VelvetToroyashi** [comment]
This is intentional. You're mentioning the role, which is called @everyone, thusly the mention becomes `@@everyone`. You can use @everyone literally instead, though we wouldn't really recommend pinging an entire server.

**Comment by VelvetToroyashi** [comment]
(Also worth noting that embeds don't support mentions—Rather that they will show the mention, but won't ping anyone)

**Comment by Ni1kko** [comment]
Why tho surly that's an issue... pointless having @everyone mention if that is the case.

**Comment by Ni1kko** [comment]
> This is intentional. You're mentioning the role, which is called @everyone, thusly the mention becomes `@@everyone`. You can use @everyone literally instead, though we wouldn't really recommend pinging an entire server.

And i ain't pinging an entire server just member who have access to said channel, using this mainly for logging purposes 

![1](https://i.imgur.com/pMItXGC.png)

**Comment by VelvetToroyashi** [comment]
Well deciphering what I can from that, I'm assuming you're referring to the point of having the EveryoneRole to begin with? In which case its use is there for many reasons, including internal permission calculations. As for being mentionable, all roles are mentionable (have the .Mention property) so there's not really much we can do in that regard.

**Comment by ProfDoof** [comment]
If you want to mention everyone you just put @everyone, the everyone role is named @everyone and is distinct from mentioning everyone.

**Comment by Ni1kko** [comment]
> If you want to mention everyone you just put @everyone, the everyone role is named @everyone and is distinct from mentioning everyone.

Thats explained a bit better, could of just said used raw text. but still the variable should never put out two mention tags that's clearly bad code with d-sharp to get it to mention. 

Also if it don't need the `<@&964059009453785099>`  to mention why not change that variable to not ouput fucked id but instead output `@everyone` as raw text rather than the <@&chan_id>

Don't make sense why you would want to keep fucked code in a well used project

**Comment by VelvetToroyashi** [comment]
Because that's extra work for a non-issue? As we said, `EveryoneRole` is just a `DiscordRole`, and so it's not really viable to add a bunch of extra code to figure out if it is the everyone role just to spit out `@everyone` instead of the mention (which, is technically inconsistent behavior, and not appropriate for a role object). The .Mention property formats a mention for the *role*, as is mentioned in the docs. There's nothing 'fucked' about this.

**Comment by akiraveliara** [comment]
1.  The everyone role is just a `DiscordRole`. It's not special-cased or for that matter possible to special-case.
2.  There may be perfectly legitimate uses to actually mention the everyone _role_, and we won't break them. It's not a lot of effort to write `@everyone`, anyways.
3.  Tangentially, there are legitimate reasons to keep bad code, generally for compatibility.

and lastly, i'll have to ask you to calm down a little and continue this discussion in a slightly more civil manner.
